test/crypto/PrivateKeyBundle.test.ts

```diff
@@ -1,24 +1,46 @@
 import * as assert from 'assert'
-import { PrivateKey, PrivateKeyBundle } from '../../src/crypto'
-import { Wallet } from 'ethers'
+import {
+  DecodePrivateKeyBundle,
+  PrivateKey,
+  PrivateKeyBundleV1,
+  PrivateKeyBundleV2,
+} from '../../src/crypto'
+import {
+  EncryptedKeyStore,
+  LocalStorageStore,
+  storageSigRequestText,
+} from '../../src/store'
 import { hexToBytes } from '../../src/crypto/utils'
+import { newWallet } from '../helpers'
+import { DecodeContactBundle } from '../../src/ContactBundle'
 
 describe('Crypto', function () {
   describe('PrivateKeyBundle', function () {
+    it('v2 generate/encode/decode', async function () {
+      const wallet = newWallet()
+      // generate key bundle
+      const bundle = await PrivateKeyBundleV2.generate(wallet)
+      const bytes = bundle.encode()
+      const bundle2 = DecodePrivateKeyBundle(bytes)
+      expect(bundle2).toBeInstanceOf(PrivateKeyBundleV2)
+      assert.ok(bundle.equals(bundle2 as PrivateKeyBundleV2))
```

Comment:
We might be able to use a discriminated union type (by having an explicit `version` field on each class), that would let us check for which type of bundle we have without needing a type assertion. The type checker should treat the value as a `v2` if we include a line like `if (bundle.version === 1) { throw new Error }` ahead of usage

Reply:
I've never actually done it with classes, but it works great with objects. 🤞

Reply:
I added the version property but it still won't let me omit the type assertion on line 26, even if I explicitly throw when version is not 2. 🤷 I'll leave the property as it will be useful for runtime checks I think.